components/Mappings/Mappings.js

```diff
@@ -434,22 +417,15 @@ class Mappings extends Component {
 			const selected = field.fields ? this.getUsecase(field.fields, this.usecases) : 'none';
 
 			if (this.state.editable) {
-				return (
-					<select
-						name="field-usecase"
-						defaultValue={selected}
-						className={dropdown}
-						onChange={(e) => {
-							this.setMapping(fieldname, 'text', e.target.value);
-						}}
-					>
-						{Object.entries(this.usecases).map(value => (
-							<option key={value[0]} value={value[0]}>
-								{value[1]}
-							</option>
-						))}
-					</select>
-				);
+				return this.renderDropDown({
```

Comment:
It would be better if we can create a Dropdown component out of in a separate file - which can be used for Search Preview as well.

Reply:
I have already added a separate drop-down in refactoring PR we can use that components in Mappings as well.